### ISSUE TITLE
Push not working?

### ISSUE THREAD

**Issue by dnish** [jagi/meteor-astronomy#358]
Hi,
I'm having some issues with push, it seems like Astronomy doesn't add the objects to the array. This is my Astronomy class:

```
Chat = Astro.Class({
    name: 'Chat',
    collection: Chats,
    fields: {
        user1: "string",
        user2: "string",
        createdAt: "date",
        messages: {
            type: 'number',
            default: 0
        },
        isTyping: {
            type:'array',
            default: function() { return []; }
        },
        lastActivity: "date"
    }
}
```

If a user is typing, I'll execute the method "isTyping":

```
'isTyping':function() 
{
   var chat = Chat.findOne({_id:chatId,$or:[{user1:this.userId},{user2:this.userId}]});
        if(!chat) return false;

        chat.push('isTyping',
            {
                userId:this.userId,
                lastActivity: new Date()
            });

        chat.save();
}
```

If I execute "getModified" after push, I only get <code>{}</code> as return value, <code>chat.validate()</code> returns <code>true</code>. Any idea if I did something wrong here?

**Comment by lukejagodzinski** [comment]
Have you tried defining field as?

``` js
isTyping: {
  type:'array',
  nested: 'object'
  default: function() { return []; }
},
```

and what's inside `chat.isTyping` after using push method?

**Comment by dnish** [comment]
If I add <code>nested:'object'</code> my whole applications crashes with:

``` ^
TypeError: Cannot read property 'nested' of undefined
    at new ObjectField (packages/jagi_astronomy/lib/modules/fields/types/object_field.js:8:1)
    at new ArrayField (packages/jagi_astronomy/lib/modules/fields/types/array_field.js:12:1)
    at packages/jagi_astronomy/lib/modules/fields/init_definition.js:181:1
    at Function._.each._.forEach (packages/underscore/packages/underscore.js:142:1)
    at Function.onInitDefinitionFields (packages/jagi_astronomy/lib/modules/fields/init_definition.js:177:1)
    at packages/jagi_astronomy/lib/modules/core/classes.js:27:1
    at packages/jagi_astronomy/lib/modules/core/events.js:68:1
    at Array.forEach (packages/es5-shim/.npm/package/node_modules/es5-shim/es5-shim.js:417:1)
    at Function._.each._.forEach (packages/underscore/packages/underscore.js:134:1)
    at Object._.extend.each (packages/jagi_astronomy/lib/modules/core/events.js:64:1)
Exited with code: 8
```

I don't know if it has something to do with the issue, but the initial chat document has "isTyping: NULL" on the server side. So I get on the client side via <code>Chat.find().fetch() -> isTyping: []</code>, as defined in defaults.

**Comment by dnish** [comment]
Yep, that's the issue. The problem is that <code>isTyping</code> is null on the server side. If I change the document manually to <code>isTyping:[]</code>, Astronomy push works fine with it. I've to say that I've added <code>default: function() { return []; }</code> after the first documents have been created on my application. Just thought Astronomy will automatically set it as an array with 0 length on the server, if the type is defined as array.

**Comment by lukejagodzinski** [comment]
Ah yes it's a problem with Astronomy 1.0 that was fixed in 2.0.

**Comment by dnish** [comment]
Ahh okay, good to know. So I'll close this issue.
